src/app/education-feature/education-form/education-form.component.spec.ts

```diff
@@ -1,25 +1,165 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
-
+import {ComponentFixture, TestBed, fakeAsync, tick} from '@angular/core/testing';
 import { EducationFormComponent } from './education-form.component';
+import {EducationComponent} from "../education/education.component";
+import {BrowserModule, By} from "@angular/platform-browser";
+import {FormsModule, ReactiveFormsModule} from "@angular/forms";
+import { RouterTestingModule } from '@angular/router/testing';
+import {EffectsModule} from "@ngrx/effects";
+import {EducationEffects} from "../../store/effects/education.effects";
+import {StoreModule} from "@ngrx/store";
+import {educationReducer} from "../../store/reducers/education.reducers";
+import { HttpClientModule } from '@angular/common/http';
+import {Router} from "@angular/router";
+import {Location} from "@angular/common";
+
+
 
 describe('EducationFormComponent', () => {
   let component: EducationFormComponent;
   let fixture: ComponentFixture<EducationFormComponent>;
+  let el: HTMLElement;
+  let location: Location;
+  let router: Router;
+
 
   beforeEach(async () => {
-    await TestBed.configureTestingModule({
-      declarations: [ EducationFormComponent ]
-    })
-    .compileComponents();
-  });
-
-  beforeEach(() => {
-    fixture = TestBed.createComponent(EducationFormComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
-  });
-
-  it('should create', () => {
-    expect(component).toBeTruthy();
-  });
+    TestBed.configureTestingModule({
+      declarations: [
+        EducationComponent,
+        EducationFormComponent,
+      ],
+      imports: [
+        BrowserModule,
+        FormsModule,
+        ReactiveFormsModule,
+        RouterTestingModule.withRoutes(
+          [ { path: 'educationform', component: EducationFormComponent },
+            { path: 'educationform/:id', component: EducationFormComponent },]
+        ),
+        HttpClientModule,
+        EffectsModule.forRoot([ EducationEffects]),
+        EffectsModule.forFeature(),
+        StoreModule.forRoot({educations: educationReducer}),
+      ]
+    }).compileComponents().then(() => {
+      fixture = TestBed.createComponent(EducationFormComponent);
+      component = fixture.componentInstance;
+    });
+     router = TestBed.get(Router);
+     location = TestBed.get(Location);
+   });
+
+
+  describe('show edit form', () => {
+    // it('mode should be edit', function () {
+    //   fixture.detectChanges();
+    //   expect(component.mode).toEqual('edit')
+    // });
+  })
+
+  describe('show add form', () => {
+    it('mode should be add', function () {
+      fixture.detectChanges();
+      expect(component.mode).toEqual('add')
+
+    });
+  })
+  describe('form validation', () => {
+
+    it('form should be invalid',  (() => {
+        component.educationForm.controls['id'].setValue('');
```

Comment:
I would recommend writing all your test in the aaa pattern (Arrange, Act and Assert). it makes it more readable.